src/GNDStk/v1.9/containers/Values.hpp

```diff
@@ -81,21 +81,27 @@ class Values : public Component<Values,true> {
    // ------------------------
 
    // length
-   const auto &length() const
+   const std::optional<Integer32> &
+   length() const
     { return content.length; }
-   auto &length()
+   std::optional<Integer32> &
+   length()
     { return content.length; }
 
    // start
-   const auto &start() const
+   const Integer32
+   start() const
     { return content.start.value(); }
-   auto &start()
+   Integer32
+   start()
     { return content.start.value(); }
 
    // valueType
-   const auto &valueType() const
+   const UTF8Text
```

Comment:
I assume that UTF8Text is a std::string?

I think you're missing a reference '&' in the return type of this function and the next one. The other functions of this type seem to include the proper character, these seem to be the only exception so far.

Reply:
`UTF8Text` is aliased to `std::string`. I use the former only because it's what they say in the JSON files. I do the same with regards to `Integer32` and `Float64`. The alternative would be to do more by-hand modifications, to the JSON files, than what we already do.

The inconsistencies in terms of constness and reference-ness, in places that use `Defaulted`, is due to the issue we spoke about in person. Everything can (and in my opinion should) be made consistent in the autogenerated files, if we give Python a correct understanding of class `Defaulted`, in the same fashion that Python understands `std::optional`. Without that fundamental fix, we'll always need to use `value()` in `Defaulted` objects, and getting correct semantics everywhere will be a game of whack-a-mole. So let's think in terms of doing something with Python.

Reply:
Another option would be to have the autogenerator automatically recognize `Integer32` as `int`, and so forth, so that the type aliases aren't even needed in GNDStk. A question would be whether the GNDS standard will ever *really* do something that makes the "32" and "64" and "UTF8" important in some way. Languages such as XML are very careful to define how they encode things in a very precise way. For GNDS, I'm guessing that it's not really important that an `Integer32` go into a 32-bit integer; if integers are 64 bits instead, that should obviously be OK. Doubles are typically 64 bits on modern machines, but don't need to be. "UTF8" is a specific thing, but I'm guessing that a C++ `std::string` will suffice for whatever someone puts into a GNDS "UTF8" string. We should probably have a discussion sometime about whether the GNDS standard's stated specificity, with these things, is actually very important or not.